src/components/Settings/PhotosSourceLocationsSettings.vue

```diff
@@ -0,0 +1,128 @@
+<!--
+ - @copyright Copyright (c) 2019 John Molakvoæ <skjnldsv@protonmail.com>
+ -
+ - @author John Molakvoæ <skjnldsv@protonmail.com>
+ -
+ - @license AGPL-3.0-or-later
+ -
+ - This program is free software: you can redistribute it and/or modify
+ - it under the terms of the GNU Affero General Public License as
+ - published by the Free Software Foundation, either version 3 of the
+ - License, or (at your option) any later version.
+ -
+ - This program is distributed in the hope that it will be useful,
+ - but WITHOUT ANY WARRANTY; without even the implied warranty of
+ - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ - GNU Affero General Public License for more details.
+ -
+ - You should have received a copy of the GNU Affero General Public License
+ - along with this program. If not, see <http://www.gnu.org/licenses/>.
+ -
+ -->
+
+<template>
+	<div class="photos-locations">
+		<ul>
+			<PhotosFolder :path="photosSourceFolder" :root-folder-label="t('photos', 'All folders')" :root-folder-icon="FolderMultiple" />
+			<!-- TODO: uncomment when SEARCH on multiple folders is implemented. -->
+			<!-- <li v-for="(source, index) in photosSourceFolder"
+				:key="index">
+				<PhotosFolder :path="source"
+					:can-delete="photosSourceFolder.length !== 1"
+					:root-folder-label="t('photos', 'All folders')"
+					@remove-folder="removeSourceFolder(index)" />
+			</li> -->
+		</ul>
+
+		<NcButton :aria-label="t('photos', 'Choose a source Photos for the timelines')"
+			@click="debounceAddSourceFolder">
+			<!-- TODO: uncomment when SEARCH on multiple folders is implemented. -->
+			<!-- <template #icon>
+				<Plus :size="20" />
+			</template> -->
+			{{ t('photos', 'Choose a different folder') }}
+		</NcButton>
+	</div>
+</template>
+
+<script>
+import debounce from 'debounce'
+import { defineComponent } from 'vue'
+
+import FolderMultiple from 'vue-material-design-icons/FolderMultiple.vue'
+
+import { NcButton } from '@nextcloud/vue'
+import { getFilePickerBuilder } from '@nextcloud/dialogs'
+import { translate as t } from '@nextcloud/l10n'
+
+import PhotosFolder from './PhotosFolder.vue'
+
+export default defineComponent({
+	name: 'PhotosSourceLocationsSettings',
```

Comment:
Feels like `PhotosSourceLocationsSettings` and `PhotosUploadLocationsSettings` could be merged into a  `PhotosPickLocationsSettings` and a few props for simplicity and avoiding duplicate code :)

Reply:
Beside the CSS and the `openFilePicker` method, they are rather different, and keeping them separated avoid having if/else conditions everywhere.

Reply:
The main part, `PhotosFolder` is already shared.